### PR TITLE
Locations

### PR DESCRIPTION
**Pull request by divyap1**
Create mail event front end

### CODE REVIEW

File: app/controllers/company_routes_controller.rb [hunk 1]
```diff
@@ -59,7 +60,8 @@ def create
 
   def destroy
     if @company_route.update(discontinued: true)
-      CreateDiscontinueEvent.new(@company_route).call
+      @last_event = Event.last
```
Comment (Recursively) [inline app/controllers/company_routes_controller.rb]: same here

File: app/controllers/company_routes_controller.rb [hunk 2]
```diff
@@ -73,7 +75,8 @@ def update
       if @company_route.update(company_route_params)
         DeliveryTime.where(company_route: @company_route).delete_all
         @company_route.calculate_delivery_times
-        CreateUpdateCostEvent.new(@company_route).call
+        @last_event = Event.last
```
Comment (Recursively) [inline app/controllers/company_routes_controller.rb]: &here

File: app/controllers/create_mails_controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    @new_mail = CreateMail.new
+    @locations = Location.select(:city, :country, :id).distinct.order('country, city')
+  end
+
+  def create
+    if params[:create_mail][:priority].blank?
+      redirect_back_with_alert('Priority must be specified')
+    else
+      @create_mail_event = CreateMail.new(mail_params)
+      byebug
```
Comment (Recursively) [inline app/controllers/create_mails_controller.rb]: byebug :(

File: app/controllers/create_mails_controller.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      @create_mail_event.find_best_route!
+      byebug
```
Comment (Recursively) [inline app/controllers/create_mails_controller.rb]: 2

File: app/controllers/create_mails_controller.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      if @create_mail_event.company_routes.present?
+        @create_mail_event.update_attributes(
+        revenue: @create_mail_event.calculate_revenue,
+        expenditure: @create_mail_event.calculate_expenditure
+        )
+        byebug
```
Comment (Recursively) [inline app/controllers/create_mails_controller.rb]: 3

File: app/controllers/create_mails_controller.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  def new
+    @new_mail = CreateMail.new
+    @locations = Location.select(:city, :country, :id).distinct.order('country, city')
+  end
+
+  def create
+    if params[:create_mail][:priority].blank?
+      redirect_back_with_alert('Priority must be specified')
+    else
+      @create_mail_event = CreateMail.new(mail_params)
```
Comment (Recursively) [inline app/controllers/create_mails_controller.rb]: does this need to be an ivar?

File: app/controllers/create_mails_controller.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      byebug
+      @create_mail_event.find_best_route!
+      byebug
+      if @create_mail_event.company_routes.present?
+        @create_mail_event.update_attributes(
+        revenue: @create_mail_event.calculate_revenue,
```
Comment (Recursively) [inline app/controllers/create_mails_controller.rb]: indent these 2 lines

File: app/controllers/kps_routes_controller.rb [hunk 8]
```diff
@@ -18,7 +18,8 @@ def create
     elsif @kps_route.save!
       @company_route = CompanyRoute.find(params[:kps_route][:company_id])
       @company_route.update!(kps_route: @kps_route)
-      CreateUpdatePriceEvent.new(@kps_route).call
+      @last_event = Event.last
```
Comment (Recursively) [inline app/controllers/kps_routes_controller.rb]: here

File: app/controllers/kps_routes_controller.rb [hunk 9]
```diff
@@ -30,7 +31,8 @@ def edit
 
   def update
     if @kps_route.update(kps_route_price_update_params)
-      CreateUpdatePriceEvent.new(@kps_route).call
+      @last_event = Event.last
```
Comment (Recursively) [inline app/controllers/kps_routes_controller.rb]: here

File: app/views/application/_sidebar_actions.html.erb [hunk 10]
```diff
@@ -14,14 +14,13 @@
...
-    <li><a href="#">Total Business Profits</a></li>
     <li><%= link_to "Amount of Mail", amount_of_mail_path %></li>
     <li><a href="#">Another great thing</a></li>
   </ul>
 </div>
 
 <% if current_user.manager %>
 <div class="sidebar-action">
-  <%= link_to "Browse Past Events", event_path(0) %>
+  <a href="/events">Browse Past Events</a>
```
Comment (Recursively) [inline app/views/application/_sidebar_actions.html.erb]: Use a link_to here rather than plan html

Comment (hc09141) [inline app/views/application/_sidebar_actions.html.erb]: Should probably stay as Event rather than Time

**Pull request merged**